### PR TITLE
Added free text date fields

### PR DESCRIPTION
**Pull request by abdusahin**
Hi @xnejp03, would you mind having a look at this PR, it includes free text inputs for date entries.
Copied from some hmrc other projects, so it'd be good to have it in one central place, not sure whether all stylings are needed though.
@benaveryee, these are the changes for free text date entries I mentioned in the morning, I'd suggest you review them for sanity check.

### CODE REVIEW

File: src/main/twirl/uk/gov/hmrc/play/views/helpers/dateInputText.scala.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
+@*
```
Comment (xnejp03) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateInputText.scala.html:1]: don't think we need this file, a normal @input should be used, like in the existing date helpers

File: src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsContent.scala.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+@import uk.gov.hmrc.play.mappers.DateFields._
+@import play.api.i18n._
+@import views.html.helper._
+@import uk.gov.hmrc.play.mappers.DateFormatSymbols
+
+@dayElem = @{ fieldName + "." + day }
+@dayInputId = @{ fieldName + "-" + day }
+
+@dateInputText(
```
Comment (xnejp03) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsContent.scala.html:26]: a normal @input should be used, like in the existing date helpers

File: src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsContent.scala.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+'_inputId -> dayInputId,
+'_label -> Messages("date.fields.day"),
+'_labelClass -> "label--inline-tight",
+'_inputClass -> "input--xxsmall form-field--inline push--left",
+'_maxFieldLength -> 2, 'placeHolder ->"DD"
+)
+@monthElem = @{ fieldName + "." + month }
+@monthInputId = @{ fieldName + "-" + month }
+
+@dateInputText(
```
Comment (xnejp03) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsContent.scala.html:37]: a normal @input should be used, like in the existing date helpers

File: src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsContent.scala.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+'_inputId -> monthInputId,
+'_label -> Messages("date.fields.month"),
+'_labelClass -> "label--inline-tight",
+'_inputClass -> "input--xxsmall form-field--inline push--left",
+'_maxFieldLength -> 2, 'placeHolder ->"MM"
+)
+@yearElem = @{ fieldName + "." + year }
+@yearInputId = @{ fieldName + "-" + year }
+
+@dateInputText(
```
Comment (xnejp03) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsContent.scala.html:48]: a normal @input should be used, like in the existing date helpers

File: src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsContent.scala.html [hunk 5]
```diff
@@ -0,0 +1,55 @@
+@*
```
Comment (xnejp03) [inline src/main/twirl/uk/gov/hmrc/play/views/helpers/dateFieldsContent.scala.html:1]: maybe a more conventient name for this? something to say that it includes input fields for all fields

**Comment by feedmypixel** [comment]
@trasahin whats happened with this PR?

**Comment by abdusahin** [comment]
Hi Ben,

That looks like an old PR and Peter has left since then, you might need to contact BTA team, who has ownership of it.

**Comment by feedmypixel** [comment]
@trasahin thanks. FYI the frontend team also has part ownership of `play-ui` is this work something you still require in `play-ui`? There are a few comments from Petr that need addressing if so.

**Comment by feedmypixel** [comment]
This is over a year old - please reopen if still needed

**Pull request closed without merging**